### PR TITLE
functional-tester/etcd-tester: silent grpclog, get revs, sleep longer

### PR DESCRIPTION
**Pull request by gyuho**

### CODE REVIEW

File: tools/functional-tester/etcd-tester/tester.go [hunk 1]
```diff
@@ -85,13 +85,22 @@ func (tt *tester) runLoop() {
 				s.Cancel()
 			}
 
-			log.Printf("etcd-tester: [round#%d case#%d] waiting 5s for pending PUTs to be committed across cluster...", i, j)
-			time.Sleep(5 * time.Second)
+			// TODO: sleep shorter or longer per failure case
+			log.Printf("etcd-tester: [round#%d case#%d] waiting 30s for pending PUTs to be committed across cluster...", i, j)
+			time.Sleep(30 * time.Second)
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: i would suggest to figure out why test exactly we need to sleep longer and put this into its recovery phase.

**Comment by gyuho** [comment]
/cc @xiang90 

Tested and checked that grpc's noisy logs are silenced.

Please review. Once this merged, I will start testing in google cloud.

Thanks,

**Comment by gyuho** [comment]
/cc @xiang90 

Just fixed some looping logic. PTAL. Thanks.

File: tools/functional-tester/etcd-tester/tester.go [hunk 2]
```diff
@@ -85,13 +85,36 @@ func (tt *tester) runLoop() {
...
+				revs, err := tt.cluster.checkRevision()
+				if err != nil {
+					if err := tt.cleanup(i, j); err != nil {
+						log.Printf("etcd-tester: [round#%d case#%d.%d] cleanup error: %v", i, j, k, err)
+						return
+					}
+					continue
+				}
+
+				if ok = checkHash(revs); ok {
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: let's print out the checking succeeded in the ok case.

File: tools/functional-tester/etcd-tester/tester.go [hunk 3]
```diff
@@ -85,13 +85,36 @@ func (tt *tester) runLoop() {
...
+					break
+				}
+			}
+
+			if !ok {
+				if err := tt.cleanup(i, j); err != nil {
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: print out the rev checking failed.

File: tools/functional-tester/etcd-tester/tester.go [hunk 4]
```diff
@@ -152,9 +175,28 @@ func (s *Status) setCase(c int) {
 	s.Case = c
 }
 
-// checkConsistency stops the cluster for a moment and get the hashes of KV storages.
-func (c *cluster) checkConsistency() error {
-	hashes := make(map[string]uint32)
+func (c *cluster) checkRevision() (map[string]int64, error) {
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: we should call get getRevisions.

File: tools/functional-tester/etcd-tester/tester.go [hunk 5]
```diff
@@ -152,9 +175,28 @@ func (s *Status) setCase(c int) {
...
+			return nil, err
+		}
+		cancel()
+		revs[u] = resp.Header.Revision
+	}
+	return revs, nil
+}
+
+// checkKVHash stops the cluster for a moment and get the hashes of KV storages.
+func (c *cluster) checkKVHash() error {
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: we should call this getKVHash

**Comment by gyuho** [comment]
@xiang90 Just addressed. PTAL. Thanks!

File: tools/functional-tester/etcd-tester/tester.go [hunk 6]
```diff
@@ -152,44 +182,55 @@ func (s *Status) setCase(c int) {
 	s.Case = c
 }
 
-// checkConsistency stops the cluster for a moment and get the hashes of KV storages.
-func (c *cluster) checkConsistency() error {
-	hashes := make(map[string]uint32)
+func (c *cluster) getRevision() (map[string]int64, error) {
+	hashes := make(map[string]int64)
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: hashes -> revs?

File: tools/functional-tester/etcd-tester/tester.go [hunk 7]
```diff
@@ -152,44 +182,55 @@ func (s *Status) setCase(c int) {
...
 	}
-	return nil
+	return hashes, nil
 }
 
-// checkConsistency returns true if all nodes have the same KV hash values.
-func checkConsistency(hashes map[string]uint32) bool {
-	var cv uint32
-	isConsistent := true
+func checkHash(hashes map[string]int64) bool {
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: checkHash -> isSameValueInMap

**Comment by gyuho** [comment]
Just fixed. PTAL. Thanks.

File: tools/functional-tester/etcd-tester/tester.go [hunk 8]
```diff
@@ -85,21 +84,52 @@ func (tt *tester) runLoop() {
...
+			for k := 0; k < 5; k++ {
+				time.Sleep(time.Second)
+				log.Printf("etcd-tester: [round#%d case#%d.%d] checking current revisions...", i, j, k)
+				revs, err := tt.cluster.getRevision()
+				if err != nil {
+					if err := tt.cleanup(i, j); err != nil {
+						log.Printf("etcd-tester: [round#%d case#%d.%d] cleanup error: %v", i, j, k, err)
+						return
+					}
+					continue
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: we need to print out something before the continue. probably failed to get revisions (%v), err ?

**Comment by xiang90** [comment]
LGTM

File: tools/functional-tester/etcd-tester/tester.go (hunk 8 above)
Comment (gyuho) [inline tools/functional-tester/etcd-tester/tester.go]: Just added. Will merge. Thanks!

**Pull request merged**